src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1325,7 +1325,7 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
 
                         ZeroPaddedSpan callDataSlice = env.InputData.SliceWithZeroPadding(src, (int) length);
                         vmState.Memory.Save(in dest, callDataSlice);
-                        if (_txTracer.IsTracingInstructions) if(length > UInt256.Zero) _txTracer.ReportMemoryChange((long) dest, callDataSlice.ToArray());
+                        if (_txTracer.IsTracingInstructions) if(length > UInt256.Zero) _txTracer.ReportMemoryChange((long) dest, callDataSlice);
```

Comment:
Calldata span passed to an externally controlled interface.

Reply:
ReadOnlySpan<> ?